cmd/kube-scheduler/app/server.go

```diff
@@ -437,13 +438,15 @@ func makeLeaderElectionConfig(config componentconfig.KubeSchedulerLeaderElection
 	if err != nil {
 		return nil, fmt.Errorf("unable to get hostname: %v", err)
 	}
+	// add a uniquifier so that two processes on the same host don't accidentally both become active
+	id := hostname + " " + string(uuid.NewUUID())
```

Comment:
wild guess: id ends up as a name in ObjectMeta. A space is not a good idea for that.

Reply:
na, maybe not. But anyway, would avoid a space in any case.

Reply:
> na, maybe not. But anyway, would avoid a space in any case.

I'm not picky.  What character that isn't valid in a fully qualified host would you like?

Reply:
a `-` should do. Your uuid should be unique enough to survive a universe time to avoid a conflict due to the character.

Reply:
> a - should do. Your uuid should be unique enough to survive a universe time to avoid a conflict due to the character.

Ok.  I just want it to be easily splittable.  I like knowing the hostname, I just like actually having one leader more.

Reply:
Without a look into the RFC, maybe -- will do?

Reply:
> Without a look into the RFC, maybe -- will do?

looks like dashes are legal and underscores aren't, so I could use `_` if we can figure out why this is failing.